Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -250,7 +244,8 @@ class TabTrayController: UIViewController, UITabBarDelegate, UITableViewDelegate
         let tab = tabManager.getTab(indexPath.item)
         let cell = tableView.dequeueReusableCellWithIdentifier(CellIdentifier) as CustomCell
         cell.titleText.text = tab.title
-        cell.background.image = tab.screenshot(size: CGSize(width: tableView.frame.width, height: TabTrayControllerUX.CellHeight))
+        let screenshotAspectRatio = tableView.frame.width / TabTrayControllerUX.CellHeight
+        cell.background.image = screenshotHelper.takeScreenshot(tab, aspectRatio: Float(screenshotAspectRatio), quality: 1)
```

Comment:
I don't think this will work. The problem is that when you open more than about 6 tabs, WebKit will start suspending the content process. And as a result, the screenshot here will be gray. We already had bug reports of that.

Why don't we screenshot, and cache, before the user navigates away from a tab? Then we always have the most recent screenshot of what the user last saw. From a live tab that we know is in good state to screenshot.